kmk/keys.py

```diff
@@ -62,112 +58,127 @@ def maybe_make_consumer_key(candidate, code, names):
 
 
 class KeyAttrDict(AttrDict):
-    def __getattr__(self, key, depth=0):
-        if depth > 1:
-            raise InfiniteLoopDetected()
+    def __getattr__(self, key):
+        return self.__getitem__(key)
 
+    def get(self, key, default=None):
         try:
-            return super(KeyAttrDict, self).__getattr__(key)
+            return self.__getitem__(key)
         except Exception:
-            pass
-
-        # Try all the other weird special cases to get them out of our way:
-        # This need to be done before or ALPHAS because NO will be parsed as alpha
-        # Internal, diagnostic, or auxiliary/enhanced keys
-
-        # NO and TRNS are functionally identical in how they (don't) mutate
-        # the state, but are tracked semantically separately, so create
-        # two keys with the exact same functionality
-        if key in ('NO', 'XXXXXXX'):
-            make_key(
-                names=('NO', 'XXXXXXX'),
-                on_press=handlers.passthrough,
-                on_release=handlers.passthrough,
-            )
-        elif key in ('TRANSPARENT', 'TRNS'):
-            make_key(
-                names=('TRANSPARENT', 'TRNS'),
-                on_press=handlers.passthrough,
-                on_release=handlers.passthrough,
-            )
-        # Basic ASCII letters/numbers don't need anything fancy, so check those
-        # in the laziest way
-        elif key in ALL_ALPHAS:
-            make_key(code=4 + ALL_ALPHAS.index(key), names=(key,))
-        elif key in ALL_NUMBERS or key in ALL_NUMBER_ALIASES:
-            try:
-                offset = ALL_NUMBERS.index(key)
-            except ValueError:
-                offset = ALL_NUMBER_ALIASES.index(key)
-
-            names = (ALL_NUMBERS[offset], ALL_NUMBER_ALIASES[offset])
-            make_key(code=30 + offset, names=names)
-        elif key in ('RESET',):
-            make_key(names=('RESET',), on_press=handlers.reset)
-        elif key in ('BOOTLOADER',):
-            make_key(names=('BOOTLOADER',), on_press=handlers.bootloader)
-        elif key in ('DEBUG', 'DBG'):
-            make_key(
-                names=('DEBUG', 'DBG'),
-                on_press=handlers.debug_pressed,
-                on_release=handlers.passthrough,
-            )
-        elif key in ('BKDL',):
-            make_key(
-                names=('BKDL',),
-                on_press=handlers.bkdl_pressed,
-                on_release=handlers.bkdl_released,
-            )
-        elif key in ('GESC', 'GRAVE_ESC'):
-            make_key(
-                names=('GESC', 'GRAVE_ESC'),
-                on_press=handlers.gesc_pressed,
-                on_release=handlers.gesc_released,
-            )
+            return default
+
+    def __getitem__(self, key):
+        key = key.upper()
+        try:
+            return super(KeyAttrDict, self).__getitem__(key)
```

Comment:
I think Line 77 is just calling line 70 again on CircuitPython. This eventually gets us to line 82 only b/c line 74 catches the RuntimeError (when it should be catching a KeyError).

Reply:
Agreed. I might not have time today, but I'll look into it as soon as I can.